HyperVExtension/src/HyperVExtension/Helpers/DevSetupAgentDeploymentHelper.cs

```diff
@@ -0,0 +1,306 @@
+        ////  ps.Streams.Progress.DataAdded += (sender, e) =>
+        var result = _powerShellService.Execute(deployDevSetupAgentStatement, PipeType.DontClearBetweenStatements);
+        if (!string.IsNullOrEmpty(result.CommandOutputErrorMessage))
+        {
+            throw new DevSetupAgentDeploymentException(
+                $"Unable to deploy DevSetupAgent service to VM with Id: {_vmId} due to PowerShell error: {result.CommandOutputErrorMessage}");
+        }
+    }
+
+    public virtual string GetSourcePath(ushort architecture)
+    {
+        if (architecture == (ushort)ProcessorArchitecture.X64)
+        {
+            // TODO: Adjust path once we have the actual DevSetupAgent package.
+            return Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "Assets", "DevSetupAgent_x64.zip");
+        }
+        else
+        {
+            throw new DevSetupAgentDeploymentException(
+                $"Unable to deploy DevSetupAgent service to VM with Id: {_vmId} due to unsupported architecture: {architecture}");
+        }
+    }
+
+    public PSObject GetSessionObject(PSCredential credential)
+    {
+        var newSessionCommand = new StatementBuilder()
+            .AddCommand("New-PSSession")
+            .AddParameter("VMId", _vmId)
+            .AddParameter("Credential", credential)
+            .Build();
+
+        var result = _powerShellService.Execute(newSessionCommand, PipeType.None);
+        if (!string.IsNullOrEmpty(result.CommandOutputErrorMessage))
+        {
+            throw new DevSetupAgentDeploymentSessionException(
+                $"Unable to create remote session for VM with Id: {_vmId} due to PowerShell error: {result.CommandOutputErrorMessage}");
+        }
+
+        return result!.PsObjects.FirstOrDefault()!;
+    }
+
+    private ushort GetVMArchitechture(PSObject session)
+    {
+        var getVMArchitechtureCommand = new StatementBuilder()
+                       .AddCommand("Invoke-Command")
+                       .AddParameter("Session", session)
+                       .AddParameter("ScriptBlock", ScriptBlock.Create("(Get-CIMInstance -Class win32_processor).Architecture"))
+                       .Build();
+
+        var result = _powerShellService!.Execute(getVMArchitechtureCommand, PipeType.None);
+        if (!string.IsNullOrEmpty(result.CommandOutputErrorMessage))
+        {
+            throw new DevSetupAgentDeploymentException(
+                               $"Unable to get VM architecture for VM with Id: {_vmId} due to PowerShell error: {result.CommandOutputErrorMessage}");
+        }
+
+        var psObject = result.PsObjects.FirstOrDefault();
+        if (psObject == null)
+        {
+            throw new DevSetupAgentDeploymentException(
+                               $"Unable to get VM architecture for VM with Id: {_vmId} due to PowerShell error: No result returned");
+        }
+
+        return (ushort)psObject.BaseObject;
+    }
+
+    private readonly string _script = @"

```

Comment:
Yes, I decided to keep it here for now since it's likely there will be some changes to fix bugs or make it more reliable, etc.  and it was easier to iterate on the changes for me.